x/pairing/keeper/pairing_test.go

```diff
@@ -2039,6 +2014,134 @@ func TestExtensionAndAddonPairing(t *testing.T) {
 	}
 }
 
+// TestMixBothExentionAndAddonPairingBug checks the following scenario:
+//
+//   - The strictest policy indicates that the selected providers should have both "ext1" extension and "add1" addon
+//
+//   - There is only one provider that support both "ext1" extension and "addon" addon. There are two more providers
+//     one supports only "ext1" and the other only supports "addon".
+//
+//     The mix filters code had a bug that if the policy requests ext1+addon and there were only providers that support either
+//     "ext1" or "addon", they wouldn't necessarily be picked over "normal" providers.
+//
+//     Test result with bug: pairing includes the provider that supports both and 3 random providers
+//     Test result without bug: pairing includes the provider that supports both, the "ext1" provider, the "addon" provider
+//     and one random provider
+func TestMixBothExentionAndAddonPairingBug(t *testing.T) {
```

Comment:
it wasn't a bug just a strict requirement, let's remove the wording of bug from here :)

Reply:
done. see https://github.com/lavanet/lava/commit/7e3610f4d558155f11daf3e24fa73afa58b06eea